x-pack/plugins/observability_solution/infra/public/pages/logs/settings/indices_configuration_panel.tsx

```diff
@@ -123,6 +151,34 @@ export const IndicesConfigurationPanel = React.memo<{
           />
         )}
       </EuiCheckableCard>
+      {numberOfLogsRules > 0 && indicesFormElement.isDirty && (
+        <>
+          <EuiSpacer size="s" />
+          <EuiCallOut
+            data-test-subj="infraIndicesPanelSettingsDangerCallout"
+            size="s"
+            title={i18n.translate('xpack.infra.sourceConfiguration.logsIndicesUsedByRulesTitle', {
+              defaultMessage: 'Rules utilize the current index pattern.',
+            })}
+            color="danger"
+            iconType="warning"
+          >
+            <FormattedMessage
+              id="xpack.infra.sourceConfiguration.logsIndicesUsedByRulesMessage"
+              defaultMessage="There {isOrAre} {numberOfLogsRules} rule{plural} that rely on the current {sourceType} setting. Changing this setting may impact the execution of these rules."
```

Comment:
"that rely on this data source setting"

You could also consider less variables (making intl easier) with something like "One or more active alerting rules rely on this data source setting." - if we link to the rules that are affected, showing the number here would be less important, but it's nice to have so I leave this part up to you.